pkg/iac/terraform/state/backend/http_reader_test.go

```diff
@@ -42,14 +42,25 @@ func TestNewHTTPReader(t *testing.T) {
 				},
 			},
 			wantURL: "https://raw.githubusercontent.com/cloudskiff/driftctl/main/.dockerignore",
-			wantErr: false,
+			wantErr: nil,
+		},
+		{
+			name: "Should fail with bad status code",
+			args: args{
+				url: "https://raw.githubusercontent.com/cloudskiff/driftctl-test/main/.dockerignore",
+				options: &Options{
+					Headers: map[string]string{},
+				},
+			},
+			wantURL: "https://raw.githubusercontent.com/cloudskiff/driftctl-badprojecturl",
+			wantErr: errors.New("error in backend HTTP(s): non-200 OK status code: 404 Not Found body: \"404: Not Found\""),
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := NewHTTPReader(tt.args.url, tt.args.options)
```

Comment:
Is this test doing a real request on Github side ?

* We should never make an unit test dependent of a third party service.
* It's very unexpected that the query is done when initializing the struct within the `New` method. It makes this code hard to test. For me the request should be done in the `Read()` call. Thus we can inject a mocked http client that will return fake response. Any reason not to do that ? (It can be refactored in another PR)

Reply:
We already discussed about that with @moadibfr. I suggested to use https://github.com/h2non/gock to mock the HTTP request but we end up to the conclusion that would make the test become a integration test. Not sure about how we should test this then.

Reply:
What about creating a `HttpClient` interface and mock the `Do` method ?

Reply:
You're right, shoud I create another PR for that?

Reply:
Yeah I think it is not directly related to this one